### PR TITLE
Fir 37710 jdbc validate connection against system engine

### PR DESCRIPTION
**Pull request by stepansergeevitch**
Improvements for isValid method:
- FIR-37621 run validation for a system engine as well
- add a connection property `validate_on_system_engine`, which forces a connection to run isValid on a system engine even if it's connected to a user engine

### CODE REVIEW

File: src/main/java/com/firebolt/jdbc/connection/FireboltConnection.java [hunk 1]
```diff
@@ -438,11 +437,15 @@ public boolean isValid(int timeout) throws SQLException {
...
 			throws SQLException {
-		FireboltProperties propertiesCopy = FireboltProperties.copy(fireboltProperties);
+		HashMap<String, String> runtimeProperties = new HashMap<>(fireboltProperties.getRuntimeAdditionalProperties());
 		if (isInternalRequest) {
-			propertiesCopy.addProperty("auto_start_stop_control", "ignore");
+			runtimeProperties.put("auto_start_stop_control", "ignore");
+		}
+		var propertiesBuilder = fireboltProperties.toBuilder().runtimeAdditionalProperties(runtimeProperties);
+		if (getSessionProperties().isValidateOnSystemEngine()) {
+			propertiesBuilder.compress(false).engine(null).systemEngine(true);
```
Comment (ptiurin) [inline src/main/java/com/firebolt/jdbc/connection/FireboltConnection.java:446]: Why do we have `compress(false)` defined here? Is there a special behaviour on system engine?

File: src/main/java/com/firebolt/jdbc/connection/settings/FireboltSessionProperty.java [hunk 2]
```diff
@@ -67,6 +67,8 @@ public enum FireboltSessionProperty {
 	USER_CLIENTS("user_clients", null, String.class, "user clients", FireboltProperties::getUserClients),
 	ACCESS_TOKEN("access_token", null, String.class, "access token", p -> "***"),
 	ENVIRONMENT("environment", "app", String.class, "Firebolt environment", FireboltProperties::getEnvironment, "env"),
+	VALIDATE_ON_SYSTEM_ENGINE("validate_on_system_engine", false, Boolean.class,
+			"Whether to validate the connection on the system engine or not. Not validated by default", FireboltProperties::isValidateOnSystemEngine),
```
Comment (ptiurin) [inline src/main/java/com/firebolt/jdbc/connection/settings/FireboltSessionProperty.java:71]: I'd reword this to avoid confusion, otherwise it sounds like we're not doing validation at all. `"Whether to validate the connection on the system engine or not. By default validates on an engine currently connected. "`  or something similar.

**Review by ptiurin — APPROVED** [review]
Couple of small points, otherwise looks good.

File: src/main/java/com/firebolt/jdbc/connection/FireboltConnection.java (hunk 1 above)
Comment (stepansergeevitch) [reply src/main/java/com/firebolt/jdbc/connection/FireboltConnection.java:446]: For system engine we automatically disable compress in JDBC.
https://github.com/firebolt-db/jdbc/blob/master/src/main/java/com/firebolt/jdbc/connection/settings/FireboltProperties.java#L97
I'm not sure of the reasons, all that I can say is that system engine doesn't work with compres as tests have shown. We receive an expected end of stream, and I failed ot understand the reason behind it
